src/content/docs/en/guides/cms/cosmic.mdx

```diff
@@ -0,0 +1,306 @@
+
+Now, we will fetch our Blog Posts using the same function `getAllPosts()` as we used before, however, this time we will execute the function inside of `getStaticPaths()` . This function is called at build time and generates and renders all of your Blog Posts at once.
+
+To access our data from Cosmic, we return each post inside of our `Astro.props` . This allows us to use that data within the HTML template portion of our component and render each page.
+
+```astro
+ ---
+ // src/pages/blog/[slug].astro
+import { getAllPosts } from '../../lib/cosmic'
+import PostBody from '../../components/PostBody'
+import { Image } from 'astro:assets'
+import { parseISO, format } from 'date-fns'
+
+export async function getStaticPaths() {
+  const data = (await getAllPosts()) || []
+
+  return data.map((post) => {
+    return {
+      params: { slug: post.slug },
+      props: { post }
+    }
+  })
+}
+
+const { post } = Astro.props
+---
+
+<article class="mx-auto max-w-screen-md pt-20">
+  <section class="border-b pb-8">
+    <h1 class="text-4xl font-bold">{post.title}</h1>
+    <div class="my-8"></div>
+    <span class="flex items-center gap-x-4 font-semibold">
+      <span class="flex flex-col">
+        <span>{post.metadata.author?.title}</span>
+        <span>{format(parseISO(post.created_at), 'MM/dd/yyyy')}</span>
+      </span>
+    </span>
+  </section>
+  {
+    post.metadata.cover_image && (
+      <Image
+        src={post.metadata.cover_image.imgix_url}
+        format="webp"
+        width={1200}
+        height={675}
+        aspectRatio={16 / 9}
+        quality={50}
+        alt={`Cover image for the blog ${post.title}`}
+        class={'my-12 rounded-md shadow-lg'}
+      />
+    )
+  }
+  <PostBody content={post.metadata.content} />
+</article>
+```
+
+### Rendering markdown with react-markdown
+
+Astro allows us to be very declarative on what frameworks we integrate in our project as well as mitigate a lot if not most of the JavaScript we ship to the client, especially with a framework like React. Since we are implementing React into a single component, we avoid shipping the entire framework code.
+
+In this component, `PostBody` , react-markdown allow us to not only parse the markdown coming from Cosmic, but control the output of HTML that gets rendered by creating an object of HTML components and adding any styles or attributes to them. They are taken in as anonymous functions with the name of the actual HTML element, and we return the modified HTML we want to render.
+
+We pass the `components` object to our `ReactMarkdown` component along with the actual content passed through in the parent component handling our Blog Post HTML template.
+
+```jsx title="src/components/PostBody.astro"
```

Comment:
Are you sure this is a `.astro` file? No frontmatter and seems perfect valid JSX?

Reply:
This is indeed a mistake on my end - it is a .jsx file